ruby/vampires.rb

```diff
@@ -1,79 +1,109 @@
-puts "How many employees will be processed?"
-number_of_employees = gets.chomp.to_i
+# 4.4 re attempt from scratch timed. 
+# 45 min code block..
 
-until number_of_employees == 0 
+# inquire name print (puts)
+# inquire age and and year of birth
+#like/want garlic bread?
+#opt for company health insurance?
+ puts "How many employees will be processed?"
+  number_of_employees = gets.chomp.to_i
 
+def vamp_dect
   puts "What is your name?"
     suspect_name = gets.chomp
   puts "How old are you?"
     suspect_age = gets.chomp.to_i
   puts "What year were you born?"
     birth_year = gets.chomp.to_i
-  
-  case suspect_age
-  when suspect_age == (2017 - birth_year)
-    age_match = true
-  when suspect_age != (2017 - birth_year)
-  	age_wrong = true
-  else suspect_age != (Integer)
-    bad_data = true
+  puts "Our company cafeteria serves garlic bread. Should we order some for you?(y or n)"
+    likes_garlic = gets.chomp
+  puts "Would you like to enroll in the company’s health insurance? (y or n)"
+    health_insurance = gets.chomp
+
+
+# assign variable to 
+  # -name
+  # -age
+  # -year of birth
+  # -want garlic bread
+  # -want health insurance 
+
+# if suspect age is correct and wants garlic bread, 
+# then probably not a vampire.
+  if suspect_age == (2017 -  birth_year)
+    suspect_age = true
+  else 
+    suspect_age = false
   end
   
-  puts "Our company cafeteria serves garlic bread. Should we order some for you? (yes or no)"
-  garlic_bread = gets.chomp
-  
-  	if garlic_bread ==  "yes"
-  		likes_garlic = true
-  	elsif garlic_bread == "no"
-  		dislikes_garlic = true
-  	else 
-  	  bad_data = true 
-  	end
-
-  puts "Would you like to enroll in the company's health insurance? (yes or no)"
-    health_insurance =  gets.chomp
+  if likes_garlic == "y"
+    likes_garlic = true
+  end
+  if
+    likes_garlic =="n"
+    likes_garlic = false
+  end
   
-  	if health_insurance == "yes"
-  		wants_insurance = true
-  	elsif health_insurance == "no"
-  		waives_insurance  = true
-  	else
-  	  bad_data = true
-  	end
+  if health_insurance == "y"
+    health_insurance = true
+  end
+  if
+    health_insurance == "n"
+    health_insurance = false
+  end
 
- 
-   puts "Please name any allergies you might have. You may enter 'done' when you are finished."
+puts "Please name any allergies you might have. You may enter 'done' when you are finished."
    allergy_input = gets.chomp
-   
 while allergy_input != "done"
       if allergy_input == "sunshine"
         p "Probably a vampire."
       break
     else allergy_input = ""
-         puts "any more allergies"
+         puts "any more allergies?"
       allergy_input = gets.chomp
       end
       next
     end
-  
   if suspect_name == "Drake Cula"
```

Comment:
Looks good but consider this nesting suggestion for improved design.

Lets consider order of operations here.  I understand that the instructions may be guiding you to just put in conditions as they are mentioned.  However, we could clean this up and make the file much more efficient by analyzing our conditions for optimal calls and runtime.

If the employee is named "Drake Cula" or "Tu Fang" then we know that they are definitely a vampire. If that's the case, is there any need to even run the rest of the code?

Consider nesting your conditions and creating a strong order of operations for optimized logic and runtime.

```
if name == "Drake Cula" || name == "Tu Fang"
   "definitely a vampire"
else
  if age_check
    do something
  elsif garlic_and_insurance_check
    do something
  else
    do something
  end
end
```